File: Timesheet-Project/Timesheet.Data/Entities/ProjectTask.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Timesheet.Data.Entities
{
    public class ProjectTask : BaseEntity
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(50)]
        public string TaskName { get; set; }

        [Required]
        [StringLength(50)]
        public string TaskDescription { get; set; }
        [Required]
        public DateTime StartDate { get; set; }
        [Required]
        public DateTime EndDate { get; set; }
        [Required]
        public string CreatedBy { get; set; }

        // Foreign key   
        [Display(Name = "Project")]
        public virtual int ProjectId { get; set; }

        [ForeignKey("Id")]
        public virtual Project Projects { get; set; }
    }
}

File: Timesheet-Project/Timesheet.UI/ClientApp/src/app/_models/projecttask.ts
export class ProjectTask {
    id?: number;
    taskName?: string;
    taskDescription?: string;
    projectId?:number;
    estimationhrs?:string;
    startDate?:Date;
    endDate?:Date;
    createdBy?:string;
    createdOn?:Date;
    projects?:Project;
}

export class Project {
    id?: number;
    name?: string;
    description?: string;
    startDate?:Date;
    endDate?:Date;
    createdBy?:string;
    createdOn?:Date;
}